batch/enrichment/src/main/java/HashMapEnrichment.java

```diff
@@ -0,0 +1,112 @@
+ * information by matching them by key. This version first builds a HashMap
+ * that is distributed to {@link HashJoinP}.
+ * <p>
+ * The {@link HashJoinP} expects enrichment table on input ordinal 0 and items
+ * to enrich on all other ordinals. The edge at ordinal 0 must have {@link
+ * com.hazelcast.jet.Edge#priority(int) priority} set to -1 to ensure, that
+ * items on this edge are processed before items to enrich.
+ * <p>
+ * The {@code readTickerInfoMap} reads the items in distributed way. In order
+ * to have full copy on each node we need to broadcast and distribute to a
+ * processor with local parallelism of 1. From there, we distribute the same
+ * {@code Map} instance to all local processors. Sending locally does not copy
+ * the object, thus this will not increase memory usage.
+ * <p>
+ * The DAG is as follows:
+ * <pre>{@code
+ *                             +--------------------+
+ *                             | Read tickerInfoMap |
+ *                             +---------+----------+
+ *                                       |
+ * +-----------------+                   | TickerInfo
+ * |  Trades source  |                   |  (broadcast, distributed edge)
+ * +--------+--------+          +--------+--------+
+ *          |                   |  Squash to map  |
+ *          | Trade             +---------------+-+    (localParallelism = 1)
+ *          |  (local round+robin edge)         |
+ * +--------+--------+                          |
+ * |      Joiner     +--------------------------+
+ * +--------+--------+                    Map<ticker, tickerInfo>
+ *          |                              (local, broadcast, priority edge)
+ *          | Object[]{trade, tradeInfo}
+ *          |
+ * +--------+--------+
+ * |      Sink       |
+ * +-----------------+
+ * }</pre>
+ */
+public class HashMapEnrichment {
+
+    private static final String TICKER_INFO_MAP_NAME = "tickerInfoMap";
+
+    public static void main(String[] args) throws Exception {
+        System.setProperty("hazelcast.logging.type", "log4j");
+
+        JetInstance instance = Jet.newJetInstance();
+        Jet.newJetInstance();
+        try {
+            TickerInfo.populateMap(instance.getMap(TICKER_INFO_MAP_NAME));
+
+            DAG dag = new DAG();
+
+            Vertex tradesSource = dag.newVertex("tradesSource", GenerateTradesP::new)
+                    .localParallelism(1);
+            Vertex readTickerInfoMap = dag.newVertex("readTickerInfoMap", readMap(TICKER_INFO_MAP_NAME));
+            Vertex squashToMap = dag.newVertex("squashToMap", SquashP::new)
+                    .localParallelism(1);
+            Vertex joiner = dag.newVertex("joiner", () -> new HashJoinP<>(Trade::getTicker));
+            Vertex sink = dag.newVertex("sink", writeLogger(o -> Arrays.toString((Object[]) o)))
+                    .localParallelism(1);
+
+            dag.edge(between(readTickerInfoMap, squashToMap)
```

Comment:
`squashToMap` is an unusual name. Semantically it's the same as the standard `writeMap()`, only it writes to a local instance. So we could call it `toLocalMap` or `writeLocalMap`.